src/libraries/System.Diagnostics.Process/tests/ProcessModuleTests.cs

```diff
@@ -93,9 +94,10 @@ public void ModulesAreDisposedWhenProcessIsDisposed()
         }
 
         public static bool Is_LongModuleFileNamesAreSupported_TestEnabled
-            => PathFeatures.AreAllLongPathsAvailable() // we want to test long paths
+            => OperatingSystem.IsWindows() // it's specific to Windows
+            && PathFeatures.AreAllLongPathsAvailable() // we want to test long paths
             && !PlatformDetection.IsMonoRuntime // Assembly.LoadFile used the way this test is implemented fails on Mono
-            && OperatingSystem.IsWindowsVersionAtLeast(8); // it's specific to Windows and does not work on Windows 7
+            && RuntimeInformation.ProcessArchitecture != Architecture.X86; // for some reason it's flaky on x86
```

Comment:
We're confident this is an OS limitation rather than a product or test bug?

Reply:
My guess is that we end up taking a fallback path in the assembly loader on x86 for some reason. The assembly loader has number of fallback paths like for historic reasons to compensate for OS and compiler issues.

The assembly loader behavior that this test depends on is not guaranteed. The test is disabled on Mono for this reason. If once we enable these tests in single-file or NativeAOT modes, the test may need to be disabled for those configurations as well. 

It would be better if the test came with its own .dll and called `LoadLibrary` on it. Then it can depend on that the library has to show up in the list.

Reply:
@jkotas I've followed your suggestion and used `LoadLibrary` directly. On one of the x86 CI legs it [fails](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-57471-merge-ae7b8909a4214c0db8/System.Diagnostics.Process.Tests/1/console.3917e2ab.log?sv=2019-07-07&se=2021-09-05T15%3A30%3A46Z&sr=c&sp=rl&sig=K1mw8HK4TiYkbuFHyvtXYHYkyxJxASn5oHxZVyd38vk%3D) with `ERROR_MOD_NOT_FOUND`:

```log
    System.Diagnostics.Tests.ProcessModuleTests.LongModuleFileNamesAreSupported [FAIL]
      LoadLibrary failed with 126
      Expected: False
      Actual:   True
      Stack Trace:
        /_/src/libraries/System.Diagnostics.Process/tests/ProcessModuleTests.Windows.cs(37,0): at System.Diagnostics.Tests.ProcessModuleTests.LongModuleFileNamesAreSupported()
```

I can't repro it locally and find out which module is missing (this is just a lib with a single class). Since this test is basically best effort to test long paths support for process module file paths, would it be OK to just skip the test on `ERROR_MOD_NOT_FOUND` on x86?

Reply:
Yes, I think it is ok to skip the test when LoadLibrary fails to load the module with the long file name. My guess is that the long paths are not properly enabled on the system when it happens.